source/wp-content/themes/wporg-showcase-2022/src/site-meta-list/index.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * Block Name: Site Screenshot
+ * Description: Display a screenshot of the site.
+ *
+ * @package wporg
+ */
+
+namespace WordPressdotorg\Theme\Showcase_2022\Site_Meta_List;
+
+add_action( 'init', __NAMESPACE__ . '\init' );
+
+/**
+ * Render the block content.
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ *
+ * @return string Returns the block markup.
+ */
+function render( $attributes, $content, $block ) {
+	if ( ! isset( $block->context['postId'] ) ) {
+		return '';
+	}
+
+	$list_items = array();
+
+	$meta_fields = array(
+		array(
+			'label' => __( 'Author', 'wporg' ),
+			'key' => 'author',
+		),
+		array(
+			'label' => __( 'Country', 'wporg' ),
+			'key' => 'country',
+		),
+		array(
+			'label' => __( 'Theme', 'wporg' ),
+			'key' => 'theme',
+		),
+		array(
+			'label' => __( 'Launched at', 'wporg' ),
+			'key' => 'date_launched',
+		),
+	);
+
+	foreach ( $meta_fields as $field ) {
+		$value = get_custom_field( $field['key'], $block->context['postId'] );
+
+		if ( ! empty( $value ) ) {
+			$label = $field['label'];
+			$list_items[] = "<dt>$label</dt><dd>$value</dd>";
+		}
+	}
+
+	$terms = get_associated_terms( $block->context['postId'] );
+
+	if ( ! empty( $terms ) ) {
+		$archive_label = __( 'Archived In', 'wporg' );
+		$list_items[] = "<dt>$archive_label</dt><dd>$terms</dd>";
+	}
+
+	// Separate items into 2 different containers
+	$half = ceil( count( $list_items ) / 2 );
+	$left = array_slice( $list_items, 0, $half );
+	$right = array_slice( $list_items, $half );
+
+	$wrapper_attributes = get_block_wrapper_attributes();
+	return sprintf(
+		'<div %s><dl><div>%s</div><div>%s</div></dl></div>',
+		$wrapper_attributes,
+		join( '', $left ),
+		join( '', $right )
+	);
+}
+
+/**
+ * Retrieves a custom site field
+ *
+ * @param string $key Name of meta field.
+ * @return string
+ */
+function get_custom_field( $key, $post_id ) {
+	$values = get_post_custom_values( $key, $post_id );
+
+	if ( empty( $values ) ) {
+		return '';
+	}
+
+	return $values[0];
+}
+
+/**
+ * Get's tags and categories associated with site.
+ *
+ * @param integer $post_id
+ * @return string
+ */
+function get_associated_terms( $post_id ) {
+	$terms = array();
+	$tags = get_the_terms( $post_id, 'post_tag' );
+	$categories = get_the_terms( $post_id, 'category' );
```

Comment:
I feel like there should be a better wp way to get all the tags and categories.

Reply:
@StevenDufresne I wonder if you could use `wp_term_query()` which accepts an array of taxonomies? https://developer.wordpress.org/reference/classes/wp_term_query/

Reply:
It doesn't appear like you can filter them based on the current post. I would probably have to recreate some of the `get_the_terms` logic:

Reply:
I think you could use `wp_get_object_terms()` to do that, since it optionally takes an array of `$taxonomy` names.

`get_the_terms()` shows how to use `wp_get_object_terms()`, you could start with the guts of that. In fact if it wasn't for its buggy cache call, `get_the_terms( $post_id, array('post_tag', 'category) )` would probably work.

I'm not sure if there's a performance advantage to doing it that way though. Your way is easy to read.

Reply:
`wp_get_object_terms()` can return Terms from multiple Taxonomies, but it bypasses the caching that `get_the_terms()` has. Realistically I think it's rare that terms from multiple taxonomies would be output in the same data..